### DISCUSSION TITLE
Accessing NodeIDs of writes from prewrite callback

### DISCUSSION THREAD

**Discussion by rlabbe82** [FreeOpcUa/opcua-asyncio#1356] [Q&A] [answered]
I'm experimenting with the ua server implementation to monitor for writes to send to an remote device. Based on reviewing some examples, I'm testing with implementing a callback to monitor the prewrite callback events of the UA server and taking the nodeid references in the request_params to determine the appropriate node is being written to and what value is being written by the external client.

I assumed that I could get the NodeID from the WriteValue objects that are populated in the request_params but I'm getting errors preventing me from accessing the WriteValue object attributes even though it seems I should be able to according to the class def (https://github.com/FreeOpcUa/opcua-asyncio/blob/a2f263611ec6891e3758462d07dc548b48523939/asyncua/ua/uaprotocol_auto.py#L7364)

![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/34889968/18a2eb82-7b79-4081-8a29-5c9c9237d42a)

![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/34889968/aadf4a03-f24b-444d-8a79-cdb9f50cefbf)

Any thoughts about what I might be doing incorrect? Thanks!

**Comment by AndreasHeine** [comment, marked as answer]
NodeId

**Reply by rlabbe82** [reply]
Wow... Guess I should have not been coding so late at night. I must have typed it 5 different ways incorrectly.

Thanks for the second set of eyes. 🙏
